File: Applied Course/4.Problem Solving/5.Problems on Trees/8.Children sum property in a binary tree.c
#include <stdio.h> 
#include <stdlib.h> 
  
struct node 
{ 
    int data; 
    struct node* left; 
    struct node* right; 
}; 
// creation of node.
struct node* newNode(int data) 
{ 
  struct node* temp = (struct node*)malloc(sizeof(struct node)); 
  temp->data = data; 
  temp->left = NULL; 
  temp->right = NULL; 
  return temp; 
} 
//Function to check sum property.
int checkSumProperty(struct node* root) 
{ 
  
  int ldata = 0,  rdata = 0; 
  //If no node or single node.
  if(root == NULL || (root->left == NULL && root->right == NULL)) 
    return 1; 
  else
  { 

    if(root->left != NULL) 
      ldata = root->left->data; 
  
    if(root->right != NULL) 
      rdata = root->right->data; 
  
    if((root->data == ldata + rdata)&& checkSumProperty(root->left) && checkSumProperty(root->right)) 
      return 1; 
    else
      return 0; 
  } 
} 
  

  
int main() 
{ 
  struct node *root  = newNode(15); 
  root->left         = newNode(10); 
  root->right        = newNode(5); 
  root->left->left   = newNode(6); 
  root->left->right  = newNode(4); 
  root->right->left = newNode(3); 
  root->right->right = newNode(2); 
  if(checkSumProperty(root)) 
    printf("children sum property "); 
  else
    printf("not satisfy the children sum property "); 
  
  return 0; 
}

File: Applied Course/4.Problem Solving/5.Problems on Trees/8.Children sum property in a binary tree.py
import sys
class node: 

    def __init__(self, info): 
        self.info = info  
        self.left = None
        self.right = None 

def checkChildSum(root):
    ldata=rdata=0
    if(root==None or (root.left==None and root.right==None)):
        return True
    else:
        if(root.left!=None):
            ldata=root.left.info
        if(root.right!=None):
            rdata=root.right.info
        
        if(root.info==ldata+rdata and checkChildSum(root.left) and checkChildSum(root.right)):
            return True
        else:
            return False

if __name__=='__main__':
    root=node(15)
    root.left=node(10)
    root.right=node(5)
    inorder(root)
    if(checkChildSum(root) is True):
        print("children sum property")
    else:
        print("not satisfy children sum property")
